dashboard/src/actions/tableOfContentActions.js

```diff
@@ -0,0 +1,63 @@
+import * as TYPES from "./types";
+import API from "../utils/axiosInstance";
+
+export const fetchTOC =
+  (param, parent, callForSubData) => async (dispatch, getState) => {
+    try {
+      const endpoints = getState().apiEndpoint.endpoints;
+      console.log(endpoints);
+      const response = await API.post(
+        `${endpoints?.api?.datasets_contents}/${param}`,
+        {
+          parent: parent,
+        }
+      );
+      if (response.status === 200 && response.data && !callForSubData) {
+        dispatch({
+          type: "GET_TOC_DATA",
+          payload: response?.data,
+        });
+      } else if (response.status === 200 && response.data && callForSubData) {
+        dispatch({
+          type: "GET_SUB_DIR_DATA",
+          payload: response?.data,
+        });
+      }
+    } catch (error) {
+      return error;
+    }
+  };
+
+export const updateTableData = (data) => async (dispatch) => {
+  dispatch({
+    type: TYPES.UPDATE_TABLE_DATA,
+    payload: data,
+  });
+};
+
+export const updateSearchSpace = (data) => async (dispatch) => {
+  dispatch({
+    type: TYPES.UPDATE_SEARCH_SPACE,
+    payload: data,
+  });
+};
+
+export const updateStack = (data) => async (dispatch) => {
+  dispatch({
+    type: TYPES.UPDATE_STACK,
+    payload: data,
+  });
+};
+
+export const updateCurrData = (data) => async (dispatch) => {
+  dispatch({
+    type: TYPES.UPDATE_CURR_DATA,
+    payload: data,
+  });
+};
+export const updateTOCLoader = (data) => async (dispatch) => {
+  dispatch({
+    type: TYPES.UPDATE_TOC_LOADING,
+    payload: data,
+  });
+};
```

Comment:
I was basically keeping the code consistent with other [similar functions](https://github.com/distributed-system-analysis/pbench/blob/main/dashboard/src/actions/datasetListActions.js) in the codebase.(getFavoritedDatasets, updateFavoriteRepoNames, updateTblData).

I will make the necessary changes here though

Reply:
Yeah, I think our understanding of thunks and actions is still evolving.  According to what I've read, the `async` keyword has no effect unless and until the marked function (or something it calls) invokes `await` (that is, asynchronous functions are initially executed synchronously).  So, since there are no `await`'s here, we should remove the `async`'s to forestall any confusion.